tool/plugin/lib/build_spec.dart

```diff
@@ -51,7 +54,40 @@ class BuildSpec {
         filesToSkip = json['filesToSkip'] ?? [],
         isUnitTestTarget = (json['isUnitTestTarget'] ?? 'false') == 'true',
         isTestTarget = (json['isTestTarget'] ?? 'false') == 'true' {
-    createArtifacts();
+    // We need Dart and an IDE but...
+    artifacts.add(
+        Artifact('Dart', dartPluginVersion, 'Dart-$dartPluginVersion.zip'));
+
+    String ideFileName;
+    // ...we're not supporting canaries...
+    if (ideaVersion != 'LATEST-EAP-SNAPSHOT') {
+      double version = double.parse(ideaVersion.split('.').take(2).join('.'));
+      // ...just
+      if (ideaProduct == 'android-studio') {
+        String prefix = version < 2020 ? '$ideaProduct-ide' : ideaProduct;
```

Comment:
The oldest version we build now is 2021.1, so you can simplify this.

Reply:
Right on. I left it jic there were some internal requirements.